go.mod

```diff
@@ -76,3 +76,7 @@ replace (
 )
 
 replace vbom.ml/util => github.com/fvbommel/util v0.0.2
+
+replace github.com/miekg/dns => github.com/guillaumerose/dns-1 v1.1.5-0.20200731203003-fe9b991fbece
```

Comment:
what is the specific change in this library?
preferably not use a fork unless there is nothing else possible.

https://github.com/guillaumerose/dns-1/commit/fe9b991fbece42eb3ef3a118dc043aecf1f48d1c

Reply:
By going from https://golang.org/pkg/net/#UDPConn to https://golang.org/pkg/net/#PacketConn you asre using a more generic interface. Why was this necessary?

Reply:
is this overriding the implementation that gvisor uses?

Reply:
*gonet.UDPConn is not *net.UDPConn. I can't use a gonet.UDPConn as a *net.UDPConn, it doesn't compile. 
I opened https://github.com/miekg/dns/pull/1170

Reply:
Ah, check. I thought these were related. Confusing names ;-)

Reply:
Added comments to that PR. While positive, we need to see what can be done when this is not merged soon-ish

Reply:
Approved: https://github.com/miekg/dns/pull/1174